Xamarin.Forms.Platform.Android/FastRenderers/ButtonRenderer.cs

```diff
@@ -294,7 +299,7 @@ protected override void OnLayout(bool changed, int l, int t, int r, int b)
 				// we just need to adjust the padding so it centers vertically
 				int diff = (b - t - _imageHeight) / 2;
 				diff = Math.Max(diff, 0);
-				SetPadding(0, diff, 0, -diff);
+				UpdateContentEdge(new Thickness (0, diff, 0, -diff));
```

Comment:
I think the math here is going to need to account for the padding when it sets the content edges. If you create a Button with bottom padding and an image
```C#
new Button()
{
     Image = "coffee.png",
     BackgroundColor = Color.Green,
     Padding = new Thickness(0, 0, 0, 100)
}
```

Then on android it looks like it takes that padding and evens it out between top and bottom ultimately centering the image instead of giving it bottom padding.  If you give it top padding and not bottom padding then it starts to push the image out of the button view

I have the code below and it mostly seems to work though I think it causes the padding to not effect the button evenly since to be perfect we'd have to know the center without the padding and use that for a constant diff. Let me check around and get some additional thoughts on the best way to handle this

```C#
	protected override void OnLayout(bool changed, int l, int t, int r, int b)
		{
			if (_imageHeight > -1)
			{	
				var bottomPadding = Context.ToPixels(Element.Padding.Bottom);
				var topPadding = Context.ToPixels(Element.Padding.Top);
				// We've got an image (and no text); it's already centered horizontally,
				// we just need to adjust the padding so it centers vertically
				var diff = ((b - bottomPadding - topPadding) - t - _imageHeight) / 2;
				diff = Math.Max(diff, 0);
				UpdateContentEdge(new Thickness(0, diff, 0, -diff));
			}

			base.OnLayout(changed, l, t, r, b);
		}
```

Reply:
I think if a padding element is specified at the bottom or top then you need to ignore the diff padding "button" with the image. 
I think it is not necessary to increase the padding, if it was specified with the user.